tests/test_polynomial.py

```diff
@@ -5,141 +5,90 @@
 
 class TestPolynomial(unittest.TestCase):
 
+    def setUp(self):
+        self.p1 = Polynomial(1, 2, 3)
+        self.p2 = Polynomial(3, 2, 1)
+        self.scalar = 5
+        self.zero_poly = Polynomial()
```

Comment:
Why this change? The premise of unit tests is to be isolated, I don't really see the need for setUp there

Reply:
The thing is that the same values were used in many places, so I decided to move these values to the `setUp` section